### PR TITLE
Refactor the #page css element

### PR DESCRIPTION
**Pull request by sabrinako**
## What’s the plan?

In broad strokes:

1. Refactor all the things that are currently nested under `#page` in `_page.scss` out into explicit classes and/or components (or simply move the styles elsewhere)
2. Put `#page` styles into a dedicated layout component (`<Page>`?)
2. Delete the `#page` declaration
3. Profit

What things need to be extracted?

- [x] `h1` through `h4`
- [x] `aside`
- [x] `header` (currently underway, this becomes `.page-header`)
- [x] `.pds-tavNab` (are we nesting this to avoid using `!important`? should there be an explicit API to override styles?)
- [x] `ul.list`
- [x] `.cards`
- [x] `.card`
- [x] `.app-decoration`
- [x] `.row`
- [x] `.item-row`
- [x] `.output-pane`
- [x] `.empty-state`
- [x] `.auth-page`
- [x] `.destroyed-deployments-filter`

**This seems like a big PR but it's just a lot of shifting things around to different files! (~800 of the deletion is from deleting the `_page.scss` file) Also I'm totally open to feedback on the folder/file structure**

### CODE REVIEW

**Review by jgwhite — APPROVED** [review]
This looks good to me. I didn’t go through the diff with a fine tooth comb, but did smoke-test it and didn’t spot any regressions. The new structure seems logical 👍

File: ui/app/styles/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+.svg-patterns {
+  display: none;
+}
+
+h1,
+h2,
+h3,
+h4 {
+  padding-bottom: scale.$sm-3;
```
Comment (gregone) [inline ui/app/styles/_layout.scss:27]: these need to use margin-bottom, not padding: 

Screen Shot 2021-06-28 at 16 29 05

File: ui/app/styles/app.scss [hunk 2]
```diff
@@ -2,147 +2,34 @@
...
+@import './components/app-item';
+@import './auth-page';
+@import './components/badge';
+@import './components/navigation/breadcrumbs';
+@import './components/button';
+@import './components/card';
+@import './components/cli-hint';
+@import './components/copyable-code';
+@import './components/empty-state';
+@import './utilities/flash';
```
Comment (gregone) [inline ui/app/styles/app.scss:22]: this is much cleaner, but some of the folder structure could be improved. `utilities` should not contain components like the flash messages (even if the styles apply to an component that comes from an external addon). I'd suggest moving this under `/components`.

File: ui/app/styles/app.scss [hunk 3]
```diff
@@ -2,147 +2,34 @@
...
+@use 'pds/tokens/font';
+@use 'pds/core/typography/config' as Typography;
+@use 'utilities/animations';
+
+@import './variables';
+@import './layout';
+
+@import './components/app-card-item';
+@import './components/app-item';
+@import './auth-page';
```
Comment (gregone) [inline ui/app/styles/app.scss:14]: move this under a `/pages` directory

File: ui/app/styles/app.scss [hunk 4]
```diff
@@ -2,147 +2,34 @@
...
+@import './components/card';
+@import './components/cli-hint';
+@import './components/copyable-code';
+@import './components/empty-state';
+@import './utilities/flash';
+@import './components/navigation/footer';
+@import './components/navigation/header';
+@import './components/icon-tile';
+@import './input';
+@import './login';
```
Comment (gregone) [inline ui/app/styles/app.scss:27]: move this under a /pages directory

File: ui/app/styles/app.scss [hunk 5]
```diff
@@ -2,147 +2,34 @@
...
+@import './components/button';
+@import './components/card';
+@import './components/cli-hint';
+@import './components/copyable-code';
+@import './components/empty-state';
+@import './utilities/flash';
+@import './components/navigation/footer';
+@import './components/navigation/header';
+@import './components/icon-tile';
+@import './input';
```
Comment (gregone) [inline ui/app/styles/app.scss:26]: move under `/components`

File: ui/app/styles/app.scss [hunk 6]
```diff
@@ -2,147 +2,34 @@
...
+@import './login';
+@import './components/notifications';
+@import './onboarding';
```
Comment (gregone) [inline ui/app/styles/app.scss:29]: mover under /pages

File: ui/app/styles/app.scss [hunk 7]
```diff
@@ -2,147 +2,34 @@
...
+@import './output-pane';
```
Comment (gregone) [inline ui/app/styles/app.scss:30]: move under /components

File: ui/app/styles/app.scss [hunk 8]
```diff
@@ -2,147 +2,34 @@
...
+@import './components/navigation/page-header';
+@import './components/structure/pds-form';
+@import './components/structure/pds-tabnav';
+@import './components/status-badge';
+@import './utilities/x-toggle';
```
Comment (gregone) [inline ui/app/styles/app.scss:35]: move under /components

**Review by gregone — CHANGES_REQUESTED** [review]

**Review by gregone — APPROVED** [review]

**Pull request merged**